### PR TITLE
Add attention dropout to SelfAttention module

### PR DESCRIPTION
**Pull request by mmz33**

### CODE REVIEW

**Review by albertz — APPROVED** [review]

File: nn/attention.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ class SelfAttention(nn.Module):
   """
   Classic self attention
   """
-  def __init__(self, *, axis: nn.DimensionTag, key_dim_total, value_dim_total, num_heads: int):
+  def __init__(self, *, axis: nn.DimensionTag, key_dim_total, value_dim_total, num_heads: int, att_dropout: float):
```
Comment (albertz) [inline nn/attention.py:13]:
```suggestion
  def __init__(self, *, axis: nn.DimensionTag, key_dim_total, value_dim_total, num_heads: int, att_dropout: float = 0.):
```

File: nn/attention.py [hunk 2]
```diff
@@ -46,7 +48,7 @@ class SelfAttentionStep(nn.Module):
   """
   Auto-regressive self-attention
   """
-  def __init__(self, *, key_dim_total, value_dim_total, num_heads: int):
+  def __init__(self, *, key_dim_total, value_dim_total, num_heads: int, att_dropout: float):
```
Comment (albertz) [inline nn/attention.py:51]:
```suggestion
  def __init__(self, *, key_dim_total, value_dim_total, num_heads: int, att_dropout: float = 0.):
```

**Review by albertz — APPROVED** [review]
